libs/web3-api-provider/src/webb-provider.ts

```diff
@@ -266,6 +265,12 @@ export class WebbWeb3Provider
       );
 
       leaves = [...storedContractInfo.leaves, ...leavesFromChain.newLeaves];
+
+      // Cached the new leaves
+      await storage.set(contract.inner.address.toLowerCase(), {
+        lastQueriedBlock: leavesFromChain.lastQueriedBlock,
+        leaves,
+      });
```

Comment:
We need to validate these leaves to be right by building the tree and checking the root to be right, this is something we can iterate in a nother PR